Content.Shared/SimpleStation14/Silicon/Components/SiliconComponent.cs

```diff
@@ -0,0 +1,108 @@
+using Robust.Shared.GameStates;
+using Content.Shared.SimpleStation14.Silicon.Systems;
+using Robust.Shared.Serialization.TypeSerializers.Implementations;
+using Robust.Shared.Containers;
+using Robust.Shared.Prototypes;
+using Content.Shared.Alert;
+
+namespace Content.Shared.SimpleStation14.Silicon.Components;
+
+/// <summary>
+///     Component for defining a mob as a robot.
+/// </summary>
+[RegisterComponent, NetworkedComponent]
+public sealed partial class SiliconComponent : Component
+{
+    [ViewVariables(VVAccess.ReadOnly)]
+    public short ChargeState = 10;
+
+    [ViewVariables(VVAccess.ReadOnly)]
+    public float OverheatAccumulator = 0.0f;
+
+    /// <summary>
+    ///     The last time the Silicon was drained.
+    ///     Used for NPC Silicons to avoid over updating.
+    /// </summary>
+    /// <remarks>
+    ///     Time between drains can be specified in
+    ///     <see cref="SimpleStationCcvars.SiliconNpc"/>
+    /// </remarks>
+    public TimeSpan LastDrainTime = TimeSpan.Zero;
+
+    /// <summary>
+    ///     The Silicon's battery slot, if it has one.
+    /// </summary>
+
+    /// <summary>
+    ///     Is the Silicon currently dead?
+    /// </summary>
+    public bool Dead = false;
```

Comment:
mobstate

Reply:
This is used for determining if the silicon is drained, not if it's actually dead. Point taken, however; this should probably be renamed